src/lib/error/ErrorReport.ts

```diff
@@ -140,7 +144,29 @@ export class ErrorReport extends Error {
       stack: this.stack,
       env: this.env,
       ...(this.originalError.code ? { code: this.originalError.code } : null),
+    })
+  }
+
+  private truncateStringsFromObject(element: any, maxStrSize: number = ErrorReport.MAX_ERROR_STRING_LENGTH) {
+    if(element === null || element === undefined) {
+      return element
+    }
+    if(typeof element === 'object') {
+      Object.keys(element).forEach((key) => {
+        element[key] = this.truncateStringsFromObject(element[key], maxStrSize)
+      })
+      return element
+    }
+    if(Array.isArray(element)) {
+      element.forEach((elementFromArray, index) => {
+        element[index] = this.truncateStringsFromObject(elementFromArray, maxStrSize)
+      })
+      return element
+    }
```

Comment:
This has to be before `typeof element === 'object'`, because array is considered to be an object